From: Denis Lunev <den@openvz.org>
Date: Fri, 22 Jan 2010 15:48:26 +0000 (+0300)
Subject: ub: incorrect skb is charged in tcp_send_synack
X-Git-Tag: spawn-2.6.32~12
X-Git-Url: http://git.openvz.org/?p=linux-2.6.27-openvz;a=commitdiff_plain;h=e3f2a20e06a08febe660a1991963320b049274a9;hp=9f4210d737d237c08260a9b7e28214064e103b06

ub: incorrect skb is charged in tcp_send_synack

New one should be charged rather than old.

http://bugzilla.openvz.org/show_bug.cgi?id=987

Signed-off-by: Denis V. Lunev <den@openvz.org>
Signed-off-by: Pavel Emelyanov <xemul@openvz.org>
---

diff --git a/net/ipv4/tcp_output.c b/net/ipv4/tcp_output.c
index 6db3e0a..0d642a4 100644
--- a/net/ipv4/tcp_output.c
+++ b/net/ipv4/tcp_output.c
@@ -2259,7 +2259,7 @@ int tcp_send_synack(struct sock *sk)
 			struct sk_buff *nskb = skb_copy(skb, GFP_ATOMIC);
 			if (nskb == NULL)
 				return -ENOMEM;
-			if (ub_tcpsndbuf_charge(sk, skb) < 0) {
+			if (ub_tcpsndbuf_charge(sk, nskb) < 0) {
 				kfree_skb(nskb);
 				return -ENOMEM;
 			}
